### PR TITLE
Fix typos in lib/kernel

### PR DESCRIPTION
**Pull request by kianmeng**

### CODE REVIEW

File: lib/kernel/examples/gen_tcp_dist/src/gen_tcp_dist.erl [hunk 1]
```diff
@@ -700,7 +700,7 @@ dist_cntrlr_setup_loop(Socket, TickHandler, Sup) ->
 
 dist_cntrlr_input_setup(DHandle, Socket, Sup) ->
     link(Sup),
-    %% Ensure we don't try to put data before registerd
+    %% Ensure we don't try to put data before registers
```
Comment (bjorng) [inline lib/kernel/examples/gen_tcp_dist/src/gen_tcp_dist.erl:703]:
```suggestion
    %% Ensure we don't try to put data before we are registered
```

File: lib/kernel/src/gen_tcp_socket.erl [hunk 2]
```diff
@@ -1439,7 +1439,7 @@ handle_event({call, From}, {setopts, Opts}, State, {P, D}) ->
     %% we are in state 'recv' but are inactive is simply stored in the buffer.
     %% If activated: active: false -> once | true | N > 0
     %% We need to check if there is something in our buffers, and maybe deliver
-    %% it to its owner. This is what we do here. This should only occure
+    %% it to its owner. This is what we do here. This should only occurred
```
Comment (bjorng) [inline lib/kernel/src/gen_tcp_socket.erl:1442]:
```suggestion
    %% it to its owner. This is what we do here. This should only occur
```

File: lib/kernel/src/logger_h_common.erl [hunk 3]
```diff
@@ -396,7 +396,7 @@ do_log_to_binary(Log,Config) ->
                                      {config,FormatterConfig},
                                      {log_event,Log},
                                      {bad_return_value,String},
-                                     {catched,{C2,R2,S2}}]),
+                                     {caught,{C2,R2,S2}}]),
```
Comment (bjorng) [inline lib/kernel/src/logger_h_common.erl:399]: Don't change this spelling since it is code.

File: lib/kernel/test/logger_std_h_SUITE.erl [hunk 4]
```diff
@@ -1780,7 +1780,7 @@ rotation_opts_restart_handler(Config) ->
     {ok,#file_info{size=29}} = file:read_file_info(Log++".0.gz"),
     [_] = filelib:wildcard(Log++".*"),
 
-    %% Stop/start handler and turn off compression. Check that achives
+    %% Stop/start handler and turn off compression. Check that achieves
```
Comment (bjorng) [inline lib/kernel/test/logger_std_h_SUITE.erl:1783]:
```suggestion
    %% Stop/start handler and turn off compression. Check that archives
```

File: lib/kernel/test/socket_SUITE.erl [hunk 5]
```diff
@@ -22045,7 +22045,7 @@ api_opt_ipv6_recvpktinfo_udp(InitState) ->
 %% There seem to be some weirdness with the definition of this
 %% option, so its defined in an include file we don't include
 %% (directly or indirectly). And since some of the defines
-%% are occure in a file we *do* include (via netinet/in.h), we
+%% are occurred in a file we *do* include (via netinet/in.h), we
```
Comment (bjorng) [inline lib/kernel/test/socket_SUITE.erl:22048]:
```suggestion
%% occurred in a file we *do* include (via netinet/in.h), we
```

**Review by bjorng — CHANGES_REQUESTED** [review]

File: lib/kernel/src/logger_h_common.erl (hunk 3 above)
Comment (kianmeng) [reply lib/kernel/src/logger_h_common.erl:399]: Should this variable name tally up with the `caught_call_exception` type here, like everywhere use `caught` instead of `catched`?

See https://github.com/erlang/otp/blob/master/lib/kernel/src/erpc.erl

```erlang
-type caught_call_exception() ::
        {throw, Throw :: term()}
      | {exit, {exception, Reason :: term()}}
      | {error, {exception, Reason :: term(), StackTrace :: [stack_item()]}}
      | {exit, {signal, Reason :: term()}}
      | {error, {?MODULE, Reason :: term()}}.
```

Comment (bjorng) [reply lib/kernel/src/logger_h_common.erl:399]: I don't think this instance of `catched` has something to do with `erpc`.

@garazdawi Do you think it safe to change this instance of `catched` to `caught` or is the code (or documentation) that depends on it remaining `catched`?

Comment (garazdawi) [reply lib/kernel/src/logger_h_common.erl:399]: Should be fine to change. The code is only used when a formatter crashes and the exact keys are not documented.

File: lib/kernel/test/socket_SUITE.erl (hunk 5 above)
Comment (bjorng) [reply lib/kernel/test/socket_SUITE.erl:22048]: Please remove the "are".

Comment (kianmeng) [reply lib/kernel/test/socket_SUITE.erl:22048]: > Please remove the "are".

@bjorng Done.

**Review by bjorng — APPROVED** [review]
Added to our daily builds.

File: lib/kernel/test/socket_SUITE.erl (hunk 5 above)
Comment (zuiderkwast) [reply lib/kernel/test/socket_SUITE.erl:22048]: Shound't it be "occur" instead? Present tense...

Comment (bjorng) [reply lib/kernel/test/socket_SUITE.erl:22048]: @zuiderkwast You are right. Present tense makes more sense.

@kianmeng Please change.

Comment (kianmeng) [reply lib/kernel/test/socket_SUITE.erl:22048]: @bjorng @zuiderkwast :ok_hand:

File: lib/kernel/test/zlib_SUITE_data/zipdoc [hunk 6]
```diff
@@ -767,7 +767,7 @@ General Format of a ZIP file
             Currently, date-time width is 32-bit, but future version may
             support be 64-bit times (see flags)
 
-          CrGMTOffs     Long(signed!)   difference "local Creat. time - UTC"
+          CrGMTOffs     Long(signed!)   difference "local Create. time - UTC"
```
Comment (bjorng) [inline lib/kernel/test/zlib_SUITE_data/zipdoc:770]: This change causes three test cases in `zlib_SUITE` to fail. Please undo the change.

**Review by bjorng — APPROVED** [review]
Thanks!

**Comment by kianmeng** [comment]
:tada: :tada: :tada: :tada: :tada:
